### PR TITLE
feature/props

### PR DESCRIPTION
**Pull request by shawnmcmahon**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

- Add prop types to all components that have props passed down to them

## Issues Closed
None

## Type of change
Check the relevant option:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (clean up code)

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Opening the app to make sure all functionality still works.

### CODE REVIEW

**Comment by shawnmcmahon** [comment]
Closing. Branch too far back in history to merge. Added prop types to a new branch.

File: src/components/App/App.js [hunk 1]
```diff
@@ -5,62 +5,61 @@ import MoviePage from '../MoviePage/MoviePage';
...
-        isLoading: true,
-        currentMovie: "",
-        test: true,
-        error: false
-      }
+  constructor() {
+    super();
+    this.state = {
+      movies: [],
+      isLoading: true,
```
Comment (kaylagordon) [inline src/components/App/App.js:13]: Try to limit your properties in state to only the data you absolutely need. Could render the loading component based on any other information in state to remove the need for `isLoading`?

File: src/components/App/App.js [hunk 2]
```diff
@@ -5,62 +5,61 @@ import MoviePage from '../MoviePage/MoviePage';
...
+      currentMovie: '',
+      test: true,
+      error: false,
+    };
   }
 
   componentDidMount = () => {
     const url = 'https://rancid-tomatillos.herokuapp.com/api/v2/movies/';
-    const url2 = 'https://httpstat.us/500'
     fetch(url)
```
Comment (kaylagordon) [inline src/components/App/App.js:22]: Consider moving your API calls into a separate file.

File: src/components/App/App.js [hunk 3]
```diff
@@ -5,62 +5,61 @@ import MoviePage from '../MoviePage/MoviePage';
...
+  };
 
   checkForError = (response) => {
-    console.log(response)
+    console.log(response);
     if (!response.ok) {
       const statusCode = response.status;
-      this.setState({ error: true});
-      throw new Error(`Something went wrong, please try again. Error Code: ${statusCode}`)
+      this.setState({ error: true });
```
Comment (kaylagordon) [inline src/components/App/App.js:32]: This is similar to the `isLoading` feedback above. I wonder if storing the actual error as a string in state would be more valuable than just a boolean.

File: src/components/App/App.js [hunk 4]
```diff
@@ -74,24 +73,23 @@ class App extends React.Component {
         rating={this.state.currentMovie.movie.average_rating}
         goBack={this.backToMain}
       />
-    )
+    );
   }
 
   render() {
```
Comment (kaylagordon) [inline src/components/App/App.js:79]: I recommend keeping your `render` method as clean as possible. Consider moving the logic to another function and invoking it from within `render`

File: src/components/Movie/Movie.js [hunk 5]
```diff
@@ -1,21 +1,31 @@
...
+      </div>
+    );
+  }
 }
 
-
 export default Movie;
+
+Movie.propTypes = {
+  src: PropTypes.string,
```
Comment (kaylagordon) [inline src/components/Movie/Movie.js:28]: Are any of these props required in order for the app to run? If so, consider using the `.isRequired` property!

File: src/components/MoviePage/MoviePage.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import React from "react"
+import PropTypes from "prop-types";
 import "./MoviePage.css"
 
 const MoviePage = (props) => {
```
Comment (kaylagordon) [inline src/components/MoviePage/MoviePage.js:5]: Consider destructuring your props here to avoid the need for `this.props` every time you access a prop.

**Review by kaylagordon — COMMENTED** [review]
Overall, great work! No need to make any of the changes I recommended, but feel free to read through my comments and implement any changes you see fit. Let me know if you have any questions!

**Pull request closed without merging**